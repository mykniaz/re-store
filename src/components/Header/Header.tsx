import * as React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

interface IProps {
  addedAmount: number;
  orderTotal: number;
}

export const Header: React.SFC<IProps> = ({ addedAmount, orderTotal }) => (
  <header className="navbar navbar-dark bg-dark mb-4">
    <Link className="navbar-brand mr-2" to="/" aria-label="ReStore">ReStore</Link>
    <div>
      <span className="text-light">added {addedAmount} items ({orderTotal}$)</span>
      <Link to="/order" className="btn btn-primary ml-4" type="submit">Buy</Link>
    </div>
  </header>
);

type TOrderItems = Array<{
  id: number;
  title: string;
  count: number;
  total: number;
}>;

const mapStateToProps = (
    {
        shoppingCard: { orderItems, orderTotal },
    }: {
      shoppingCard: { orderItems: TOrderItems, orderTotal: number },
    },
    ) => {
  const reducer = (accumulator: number, order: { count: number }) => accumulator + order.count;

  return {
    orderTotal,
    addedAmount: orderItems.reduce(reducer, 0),
  };
};

export default connect(mapStateToProps)(Header);
