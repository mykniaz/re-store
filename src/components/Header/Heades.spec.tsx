import * as React from 'react';
import { Header } from './Header';
import reactTestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

test('Header render test', () => {
  const component = reactTestRenderer.create(
      <BrowserRouter>
        <Header addedAmount={3} orderTotal={1488} user={{ name: 'Test User' }}/>
      </BrowserRouter>,
    );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
