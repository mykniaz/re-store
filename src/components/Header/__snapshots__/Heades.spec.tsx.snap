// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render test 1`] = `
<header
  className="navbar navbar-dark bg-dark mb-4"
>
  <a
    aria-label="ReStore"
    className="navbar-brand mr-2"
    href="/"
    onClick={[Function]}
  >
    ReStore
  </a>
  <div>
    <span
      className="text-light"
    >
      added 
      3
       items (
      1488
      $)
    </span>
    <a
      className="btn btn-primary ml-4"
      href="/order"
      onClick={[Function]}
      type="submit"
    >
      Buy
    </a>
  </div>
</header>
`;
